### PR TITLE
Fix the GeckoProfiler API call and prefs with the latest Firefox API

### PR DESCRIPTION
**Pull request by tarekziade**
The feature was removed in https://bugzilla.mozilla.org/show_bug.cgi?id=1572337

### CODE REVIEW

**Comment by tarekziade** [comment]
wait. there's more coming. the Services.profiler.StartProfiler() function signature has changed in Firefox

File: lib/firefox/webdriver/firefoxDelegate.js [hunk 1]
```diff
@@ -104,6 +109,7 @@ class FirefoxDelegate {
       }
 
       // Must use String.raw or else the backslashes on Windows will be escapes.
+      log.info('Collecting Gecko profile in ' + destinationFilename);
```
Comment (freshstrangemusic) [inline lib/firefox/webdriver/firefoxDelegate.js]: How about a template string?

File: lib/firefox/webdriver/firefoxDelegate.js [hunk 2]
```diff
@@ -112,6 +118,7 @@ class FirefoxDelegate {
        `;
       await runner.runPrivilegedAsyncScript(script, 'Collect GeckoProfiler');
       log.info('Stop GeckoProfiler.');
+
```
Comment (freshstrangemusic) [inline lib/firefox/webdriver/firefoxDelegate.js]: undo

File: lib/firefox/webdriver/firefoxDelegate.js [hunk 3]
```diff
@@ -63,8 +64,12 @@ class FirefoxDelegate {
         ${chosenFeatures.length},${threadString},${chosenThreads.length});`;
       } else if (parameterLength === 5) {
         script = `Services.profiler.StartProfiler(${bufferSize},${interval},${featureString},${threadString});`;
+      } else if (parameterLength === 6) {
+        // XXX what about aFilters, aActiveBrowsingContextID and aDuration ?
+        script = `Services.profiler.StartProfiler(${bufferSize},${interval},${featureString});`;
       } else {
         log.error('Unknown Gecko Profiler API');
+        throw new Error("Unknown Gecko Profiler API");
```
Comment (freshstrangemusic) [inline lib/firefox/webdriver/firefoxDelegate.js]: We're using single quotes elsewhere.

File: lib/firefox/webdriver/firefoxDelegate.js [hunk 4]
```diff
@@ -63,8 +64,12 @@ class FirefoxDelegate {
         ${chosenFeatures.length},${threadString},${chosenThreads.length});`;
       } else if (parameterLength === 5) {
         script = `Services.profiler.StartProfiler(${bufferSize},${interval},${featureString},${threadString});`;
+      } else if (parameterLength === 6) {
+        // XXX what about aFilters, aActiveBrowsingContextID and aDuration ?
```
Comment (freshstrangemusic) [inline lib/firefox/webdriver/firefoxDelegate.js]: These are optional params according to https://searchfox.org/mozilla-central/rev/ea63a0888d406fae720cf24f4727d87569a8cab5/tools/profiler/gecko/nsIProfiler.idl#38

File: lib/firefox/webdriver/firefoxDelegate.js [hunk 5]
```diff
@@ -52,6 +52,7 @@ class FirefoxDelegate {
       let bufferSize = this.firefoxConfig.geckoProfilerParams.bufferSize;
 
       // Firefox 69 and above has a slightly different API for StartProfiler
+      // See https://searchfox.org/mozilla-central/source/tools/profiler/gecko/nsIProfiler.idl#38
```
Comment (freshstrangemusic) [inline lib/firefox/webdriver/firefoxDelegate.js]: This should probably be a permalink if we want to reference a specific version of the IDL

**Review by freshstrangemusic — CHANGES_REQUESTED** [review]

File: lib/firefox/webdriver/firefoxDelegate.js (hunk 4 above)
Comment (tarekziade) [reply lib/firefox/webdriver/firefoxDelegate.js]: Yeah I know. But what about them. Should we use them ? 
cc @fqueze

File: lib/firefox/webdriver/firefoxDelegate.js (hunk 5 above)
Comment (tarekziade) [reply lib/firefox/webdriver/firefoxDelegate.js]: It's mostly to get to the idl definition. Maybe we can use, so it's future proof even if it moves around:

https://searchfox.org/mozilla-central/search?q=symbol:%23StartProfiler%2C_ZN11nsIProfiler13StartProfilerEjdRK8nsTArrayI9nsTStringIcEES5_md&redirect=false

File: lib/firefox/webdriver/firefoxDelegate.js (hunk 4 above)
Comment (fqueze) [reply lib/firefox/webdriver/firefoxDelegate.js]: I think aFilters here is ${threadString} and you'll want it.

You'll want to specify aActiveBrowsingContextID if you would like the profiler to know which tab you care about. This seems desirable for browsertime, but probably unrelated to what you are trying to do. https://hg.mozilla.org/mozilla-central/rev/59139e8c6636 has examples.

I think aDuration is irrelevant here.

Comment (tarekziade) [reply lib/firefox/webdriver/firefoxDelegate.js]: Thanks @fqueze - as this point browsertime is doing everything in the same tab afaik, so I guess this is irrelevant for now

File: lib/firefox/geckoProfilerDefaults.js [hunk 6]
```diff
@@ -1,7 +1,7 @@
 'use strict';
 
 module.exports = {
-  features: 'js,stackwalk,leaf,responsiveness',
+  features: 'js,stackwalk,leaf',
```
Comment (freshstrangemusic) [inline lib/firefox/geckoProfilerDefaults.js:4]: There is a trailing comma here that wasn't here before and I don't know how well we handle that in the API.

**Review by freshstrangemusic — CHANGES_REQUESTED** [review]
~One last thing~

EDIT: LGTM

File: lib/firefox/geckoProfilerDefaults.js (hunk 6 above)
Comment (freshstrangemusic) [reply lib/firefox/geckoProfilerDefaults.js:4]: Nevermind, I misread this :)

**Review by freshstrangemusic — APPROVED** [review]

**Pull request merged**